week-2/Homework/mandatory/3-playing-computer.js

```diff
@@ -12,13 +12,19 @@
 
   Answer the following questions:
 
-  1. This program throws an error. Why? (If you can't find it, try executing it).
-  2. Remove the line that throws the error.
+  1. This program throws an error. Why? (If you can't find it, try executing it). // b is undefined//
+  2. Remove the line that throws the error. //removed//
   3. What is printed to the console?
-  4. How many times is "f1" called?
-  5. How many times is "f2" called?
-  6. What value does the "a" parameter take in the first "f1" call?
-  7. What is the value of the "a" outer variable when "f1" is called for the first time?
+  //i= 0 , a = 7 , f2 = 4
+ i= 1 , a = 8 , f1 = 9
+ i= 2 , a = 9 , f2 = 6
+ i= 3 , a = 10 , f1 = 13
+ i= 4 , a = 11 , f2 = 8//
+
+  4. How many times is "f1" called?        //2//
+  5. How many times is "f2" called?        //3//
+ . What value does the "a" parameter take in the first "f1" call?       //1//
+  7. What is the value of the "a" outer variable when "f1" is called for the first time?    //8//
```

Comment:
all correct except 6 and 7
6. a will enter as 6+1+1=8
7. a will be 6 outside

Reply:
I don't understand questions 6 and 7